pages/ekman.js

```diff
@@ -0,0 +1,282 @@
+import React, { useState } from "react";
+import { Container, Page } from "@/components/Layout/Layout.styled";
+import NavButton from "@/components/NavButton/NavButton";
+import Circle from "@/components/Circle/Circle";
+import { TbList } from "react-icons/tb";
+import {
+  CircleContainer,
+  StyledTextarea,
+} from "@/components/EmotionTextAnalysis/EmotionTextAnalysis.styled";
+
+function EmotionTextAnalysis() {
+  const [formData, setFormData] = useState(Array(10).fill(""));
+  const [emotionResult, setEmotionResult] = useState("");
+  const [language, setLanguage] = useState("en");
+  const [page, setPage] = useState(0);
+  const [predictionsState, setPredictionsState] = useState(false);
+  const [showList, setShowList] = useState(false);
+
+  const handleChange = (event) => {
+    const { name, value } = event.target;
+    setFormData({
+      ...formData,
+      [name]: value,
+    });
+  };
+
+  const handleLanguageSelect = (event) => {
+    setLanguage(event.target.options[event.target.selectedIndex].value);
+  };
+  //information to the used API
+  //   //https://rapidapi.com/symanto-symanto-default/api/ekman-emotion-analysis
+
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+    try {
+      const response = await fetch("/api/analyze-emotion", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({ formData, language: language }),
+      });
+
+      const data = await response.json();
+      setEmotionResult(data); // Update state with API response
+      setPredictionsState(true);
+    } catch (error) {
+      console.error("Error:", error);
+    }
+  };
+
+  const question = [
+    "1. What word describes your primary emotion right now?",
+    "2. How would you describe that emotion?",
+    "3. How does your body feel?",
+    "4. Is there an event, person, place, or thing that might have caused this response?",
+    "5. How did you react to that trigger?",
+    "6. How will you respond now?",
+    "7. Are you taking care of your physical well-being?",
+    "8. What is on your mind today?",
+    "9. What can you do to feel calmer?",
+    "10. What are you grateful for today?",
+  ];
+
+  const predictions =
+    emotionResult && emotionResult[0] && emotionResult[0].predictions;
+
+  const transformedPredictions =
+    predictions &&
+    predictions[0] &&
+    predictions.map((prediction) => {
+      if (prediction.prediction === "noemo") {
+        return { ...prediction, prediction: "no-emotion" };
+      } else {
+        return prediction;
+      }
+    });
```

Comment:
We could go one step further here and also use [Nullish coalescing](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing). 

``` js
const transformedPredictions = predictions?.map(prediction => {
  return prediction.prediction === "noemo"
    ? { ...prediction, prediction: "no-emotion" }
    : prediction;
}) ?? [];
```